### ISSUE TITLE
New linter: detect catch clauses that only rethrow

### ISSUE THREAD

**Issue by marco-carvalho** [clj-kondo/clj-kondo#2971]
- [ ] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**

A final catch clause whose only expression rethrows the same caught exception has no effect and adds misleading control flow.

```clojure
(try
  (work)
  (catch Exception e
    (throw e)))
```

**Describe the solution you'd like**

Add a `:useless-catch` linter, enabled at warning level by default, for a final catch clause containing only a direct rethrow of its binding. Do not warn when the catch performs another action first, throws another value, or is followed by a broader catch that changes behavior.

**Describe alternatives you've considered**

A general redundant-expression linter does not model exception matching or catch ordering. Removing any rethrowing catch without checking its position could change behavior.

**Additional context**

I am interested in submitting a focused PR with catch ordering and side-effect cases if this approach is accepted.
